npe2/manifest/contributions/_menus.py

```diff
@@ -1,27 +1,26 @@
+from collections import namedtuple
 from typing import Optional, Union
 
 from pydantic import BaseModel, Field
 
 from ..utils import Executable
 
-# # napari provides these
-# class Menu(BaseModel):
-#     key: str
-#     id: int
-#     description: str
-#     supports_submenus: bool = True
-#     deprecation_message: Optional[str]
-
-
-# napari_menus = [
-#     Menu(key="command_pallete", id=0, description="The Command Palette"),
-#     Menu(
-#         key="layers__context", id=1, description="The layer list context menu"
-#     ),
-#     Menu(
-#         key="layers__context", id=1, description="The layer list context menu"
-#     ),
-# ]
+Menu = namedtuple("Menu", "key description")
+
+# a list of valid napari menu locations that plugins can contribute too.
+# keys provided in the plugin manifest must match these strings
+napari_menus = [
+    Menu("/napari/layer_context", "Process Layer"),
+    Menu("/napari/layer_context/projections", "Make Projection"),
+    Menu("/napari/layer_context/convert_type", "Convert datatype"),
+    Menu("/napari/tools/acquisition", "Acquisition"),
+    Menu("/napari/tools/classification", "Classification"),
+    Menu("/napari/tools/measurement", "Measurement"),
+    Menu("/napari/tools/segmentation", "Segmentation"),
+    Menu("/napari/tools/transform", "Transform"),
+    Menu("/napari/tools/utilities", "Utilities"),
+    Menu("/napari/tools/visualization", "Visualization"),
```

Comment:
```suggestion
    Menu("/napari/layer_context", "Process Layer"),
    Menu("/napari/layer_context/projections", "Make Projection"),
    Menu("/napari/layer_context/convert_type", "Convert datatype"),
    Menu("/napari/tools/acquisition", "Acquisition"),
    Menu("/napari/tools/classification", "Classification"),
    Menu("/napari/tools/filters", "Filters"),
    Menu("/napari/tools/measurement", "Measurement"),
    Menu("/napari/tools/segmentation", "Segmentation"),
    Menu("/napari/tools/projection", "Projection"),
    Menu("/napari/tools/transform", "Transform"),
    Menu("/napari/tools/utilities", "Utilities"),
    Menu("/napari/tools/visualization", "Visualization"),
```
Two things (open for discussion):
1. Denoising / background-removal / etc. are calling for a "Filters" menu. It could also be called "Process" like in other related software.
2. I'm adding the "Projection" menu to the main menu in parallel to the right-click menu. Reason: Users don't see that that right-click menu exists. The main menu is visible. Thus, projections become more accessible. Alternatively: Projections and transforms go well together in one menu.